generic/gdb.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Pavithra <pavrampu@linux.vnet.ibm.com>
+
+import os
+import re
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import distro
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class GDB(Test):
+
+    def setUp(self):
+        sm = SoftwareManager()
+        dist = distro.detect()
+        packages = ['gcc', 'dejagnu', 'flex', 'bison']
+        if dist.name == 'Ubuntu':
+            packages.extend(['g++', 'binutils-dev'])
+        elif dist.name in ['SuSE', 'redhat', 'fedora']:
+            packages.extend(['gcc-c++', 'binutils-devel'])
+        else:
+            self.fail('no packages list for your distro.')
+        for package in packages:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error("Fail to install %s required for this test." %
+                           package)
+        gdb_version = self.params.get('gdb_version', default='7.10')
+        tarball = self.fetch_asset(
+            "http://ftp.gnu.org/gnu/gdb/gdb-%s.tar.gz" % gdb_version)
+        archive.extract(tarball, self.srcdir)
+        self.srcdir = os.path.join(
+            self.srcdir, os.path.basename(tarball.split('.tar')[0]))
+        os.chdir(self.srcdir)
+        process.run('./configure', ignore_status=True, sudo=True)
+        build.make(self.srcdir)
+
+    def test(self):
+        process.run("make check-gdb", ignore_status=True, sudo=True)
+        logfile = os.path.join(self.logdir, "stdout")
+        fp = open(logfile, "r")
```

Comment:
This is keeping the file open and then avocado cannot remove the srcdir after the test:
It is a best practice to use `with` to deal with file objects, so the file is always closed after the commands inside the with....
```
with open(file, 'r') as f:
    for line in f.readlines():
...
```